src/main.cpp

```diff
@@ -3641,9 +3687,15 @@ bool ReceivedBlockTransactions(
                 } else {
                     pindex->nChainSaplingValue = boost::none;
                 }
+                if (pindex->pprev->nChainTransparentValue && pindex->nTransparentValue) {
+                    pindex->nChainTransparentValue = *pindex->pprev->nChainTransparentValue + *pindex->nTransparentValue;
```

Comment:
I suspect this will become something like (untested):
```suggestion

                // Given that all parents are BLOCK_VALID_TRANSACTIONS, we now should be
                // able to calculate this block's net transparent value, assuming the
                // block is correct.
                CAmount transparentValue = 0;
                for (auto tx : block.vtx) {
                    for (auto out : tx.vout) {
                        transparentValue += out.nValue;
                    }

                    if (view.HaveInputs(tx)) {
                        transparentValue -= view.GetTransparentValueIn(tx);
                    } else {
                        // Some of the inputs are inaccessible; this might be a block from
                        // a fork that spends coins also spent in the current chain. Fall
                        // back to directly looking up the transactions.
                        // TODO
                    }
                }
                pindex->nTransparentValue = transparentValue;
                if (pindex->pprev->nChainTransparentValue) {
                    pindex->nChainTransparentValue = *pindex->pprev->nChainTransparentValue + transparentValue;
```

along with a `CCoinsViewCache view(pcoinsTip);` before the `while (!queue.empty()) {` above.